python/pyspark/accumulators.py

```diff
@@ -109,10 +109,14 @@
 
 def _deserialize_accumulator(aid, zero_value, accum_param):
     from pyspark.accumulators import _accumulatorRegistry
-    accum = Accumulator(aid, zero_value, accum_param)
-    accum._deserialized = True
-    _accumulatorRegistry[aid] = accum
-    return accum
+    # If this certain accumulator was deserialized, don't overwrite it.
+    if aid in _accumulatorRegistry:
```

Comment:
Should it be `if aid in _accumulatorRegistry and _accumulatorRegistry[aid]._deserialized is True`
or:
```
if aid in _accumulatorRegistry:
    _accumulatorRegistry[aid]._deserialize = True
    return _accumulatorRegistry[aid]
```
To make double sure that this function always returns a deserialize version of the accum ?

Reply:
We only save deserialized accumulators (`_deserialized` is `True`) into this dict.

Reply:
That doesnt seem right because the constructor for `Accumulator` has:
```
        ...
        self._deserialized = False
        _accumulatorRegistry[aid] = self
```

Reply:
Yeah, but `_deserialize_accumulator` is only called when doing deserialzation at executors. The constructor saves accumulators in `_accumulatorRegistry` at driver.

Reply:
I see - got it :+1: